Modules/posixmodule.c

```diff
@@ -5494,7 +5494,7 @@ os__path_splitroot_ex_impl(PyObject *module, PyObject *p)
     }
 #ifdef MS_WINDOWS
     len = path.length;
-    const wchar_t *buffer = path.wide;
+    buffer = path.wide;
```

Comment:
Can we `const` the earlier `buffer` definition to avoid the warnings here?

Reply:
Ah, GitHub landed me on a partial commit, so I didn't see the rest of the change.

I'd rather define the variable as `const` if we can than cast away the `const`. If it needs to be non-const for non-Windows (though I don't see why), declare it inside the `#ifdef` (we no longer require all declarations at the top of the function, so it can go on the first assignment).